[dabzpengu] IP

dabzpengu opened the pull request:
# Gandalf

> “All we have to decide is what to do with the time that is given to us” - Gandalf the Grey

Gandalf frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
~~FAST~~ SUPER FAST to use
All you need to do is,

download it
run terminal relative to where the chatbot is

type `java -jar gandalf.jar`
let it manage your tasks **AND** expenses for you 😉
And it is FREE!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

dabzpengu committed:
renamed, greeted and exit

dabzpengu committed:
echo

dabzpengu committed:
add, list

dabzpengu committed:
can mark tasks as done or not done

dabzpengu committed:
todos, events, Deadlines

dabzpengu committed:
added textUItesting

dabzpengu committed:
handle errors

dabzpengu committed:
delete function

dabzpengu committed:
save feature

dabzpengu committed:
Merge branch 'branch-level-7'

hjuntan commented on src/main/java/GandalfException.java:
```diff
@@ -0,0 +1,5 @@
+public class GandalfException extends Exception{
```
Add whitespace after extending `Exception`.

hjuntan commented on src/main/java/GandalfException.java:
```diff
@@ -0,0 +1,5 @@
+public class GandalfException extends Exception{
+    public GandalfException(String message){
```
Add whitespace after calling constructor.

hjuntan commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,143 @@
+        }
+    }
+    public static void main(String[] args) {
+        ArrayList<Task> faster_list = new ArrayList<>(100);
```
Rename `faster_list` to `fasterList`.

hjuntan commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,143 @@
+     * @throws GandalfException
+     */
+    public static void checkCommand(String[] taskInfo, ArrayList<Task> data) throws GandalfException{
+        if(!taskInfo[0].equals("todo") && !taskInfo[0].equals("deadline") && !taskInfo[0].equals("event") && !taskInfo[0].equals("delete") && !taskInfo[0].equals("mark")&& !taskInfo[0].equals("unmark")){
```
Line wrap this long conditional statement into multiple lines.

hjuntan commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,143 @@
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.BufferedWriter;
```
Import `java.io.BufferedWriter` before `java.io.File`

hjuntan commented on src/main/java/Gandalf.java:
```diff
@@ -0,0 +1,143 @@
+        }
+        String filePath = "docs/gandalfData.txt";
+        File file = new File(filePath);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))){
```
Add whitespace after `try (...)`

dabzpengu committed:
interpret time feature and redid parsing logic

dabzpengu committed:
still in the midst of doing oop

dabzpengu committed:
extracted more OOP

dabzpengu committed:
packaged classes

dabzpengu committed:
Merge branch 'A-Packages'

dabzpengu committed:
Merge branch 'add-gradle-support'

dabzpengu committed:
run gradle with chatbot

dabzpengu committed:
j-unit test classes

dabzpengu committed:
Merge branch 'A-JUnit'

dabzpengu committed:
able to create .jar file

dabzpengu committed:
Merge branch 'A-Jar'

dabzpengu committed:
Added header comments

dabzpengu committed:
coding standard

dabzpengu committed:
implemented find feature

dabzpengu committed:
Merge branch 'A-JavaDoc'

dabzpengu committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/GandalfBot/Deadlines.java
#	src/main/java/GandalfBot/Gandalf.java
#	src/main/java/GandalfBot/Parser.java
#	src/main/java/GandalfBot/Task.java
#	src/main/java/GandalfBot/TaskList.java

dabzpengu committed:
Merge branch 'branch-level-9'

# Conflicts:
#	src/main/java/GandalfBot/Gandalf.java
#	src/main/java/GandalfBot/Task.java

Lalelulilulela commented on src/main/java/GandalfBot/Parser.java:
```diff
@@ -0,0 +1,42 @@
+    public StringBuilder[] interpret(){
+=======
+    public StringBuilder[] interpret() {
+>>>>>>> A-CodingStandard
```
Perhaps you could resolve this merge conflict?

Lalelulilulela commented on src/main/java/GandalfBot/Task.java:
```diff
@@ -0,0 +1,51 @@
+     *
+     * @param status The status of the task (true for completed, false for not completed).
+     */
+    public void markStatus(boolean status) {
```
Great method naming!

Lalelulilulela commented on src/main/java/GandalfBot/Task.java:
```diff
@@ -0,0 +1,51 @@
+     * @return The status of the task (true for completed, false for not completed).
+     */
+
+    public boolean getStatus() {
+        return this.status;
+    }
+    public String getNameOfTask() {
```
Great method naming

Lalelulilulela commented on src/main/java/GandalfBot/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+ * Class to handle operations that changes the length of the list,ie. add or delete
+ */
+public class TaskList {
+    ArrayList<Task> list;
```
Perhaps this multi-valued variable can be named in plural form?

Lalelulilulela reviewed (commented):
Great work overall!  Just need a few tweaks to comply with coding standards. Keep up the good work💪💪

dabzpengu committed:
doing checkstyle

dabzpengu committed:
Merge branch 'A-CheckStyle'

dabzpengu committed:
added gui using fxml

dabzpengu committed:
Merge branch 'branch-level-10-fxml'

dabzpengu committed:
Assert code logic for add, mark, unmark and delete

To ensure that assumptions for each method holds

Insert assert statements at certain parts of the code

dabzpengu committed:
Merge pull request #2 from dabzpengu/heads/A-Assertions

Assert code logic for add, mark, unmark and delete

dabzpengu committed:
created more OOP for commands, and its constructor needs to be
flexible

improve use of OOP and since different command types uses
different string parameters, varargs is used to
allow more flexibility for polymorphism

dabzpengu committed:
Merge pull request #3 from dabzpengu/heads/A-CodeQuality

created more OOP for commands, and its constructor needs to be flexible

dabzpengu committed:
Merge branch 'master' of https://github.com/dabzpengu/ip

dabzpengu committed:
Reorganized package structure and add extension

commands and tasktypes are placed in their own packages.
extension to allow user to track their expenses

restructured package to improve readability.

dabzpengu committed:
Restructured package and added extension

placed commands and tasktypes into their own packages.
Allows user to track their expenses

improve readability

dabzpengu committed:
Merge branch 'D-Expenses'

dabzpengu committed:
Fixed GUI to be able to process the commands

dabzpengu committed:
fixing GUI

dabzpengu committed:
added ui screenshot

dabzpengu committed:
Merge branch 'fixing-GUI'

dabzpengu committed:
created user guide

dabzpengu committed:
files update

dabzpengu committed:
Merge branch 'A-UserGuide'

dabzpengu committed:
removed data folder

dabzpengu committed:
fixed gradle

dabzpengu committed:
edited UG due to spelling error
